client/my-sites/media-library/media-date-range.jsx

```diff
@@ -0,0 +1,344 @@
+/** @format */
+
+/**
+ * External dependencies
+ */
+import React, { Component } from 'react';
+import { bindAll, noop } from 'lodash';
+import { DateUtils } from 'react-day-picker';
+import classNames from 'classnames';
+import PropTypes from 'prop-types';
+
+/**
+ * Internal dependencies
+ */
+import Button from 'components/button';
+import Popover from 'components/popover';
+import DatePicker from 'components/date-picker';
+import Gridicon from 'gridicons';
+import { localize } from 'i18n-calypso';
+import FormLabel from 'components/forms/form-label';
+import FormTextInput from 'components/forms/form-text-input';
+
+export class MediaDateRange extends Component {
+	static propTypes = {
+		onDateSelect: PropTypes.func,
+		onDateCommit: PropTypes.func,
+	};
+
+	static defaultProps = {
+		onDateSelect: noop,
+		onDateCommit: noop,
+	};
+
+	static dateFormat = 'MM/DD/YYYY';
+
+	constructor( props ) {
+		super( props );
+
+		this.state = {
+			popoverVisible: false,
+			oldStartDate: '',
+			oldEndDate: '',
+			startDate: this.props.startDate || this.props.moment().subtract( 1, 'months' ),
+			endDate: this.props.endDate || this.props.moment(),
+			oldDatesSaved: false,
+			inputFromDate:
+				this.dateToHumanReadable( this.props.startDate ) ||
+				this.dateToHumanReadable( this.props.moment().subtract( 1, 'months' ) ),
+			inputToDate:
+				this.dateToHumanReadable( this.props.endDate ) ||
+				this.dateToHumanReadable( this.props.moment() ),
+		};
+
+		bindAll( this, [
+			'togglePopover',
+			'closePopover',
+			'onSelectDate',
+			'revertDates',
+			'commitDates',
+			'handleInputChange',
+			'handleInputBlur',
+		] );
+
+		this.startButtonRef = React.createRef();
+	}
+
+	togglePopover() {
+		this.setState( {
+			popoverVisible: ! this.state.popoverVisible,
+			inputFromDate: this.dateToHumanReadable( this.state.startDate ),
+			inputToDate: this.dateToHumanReadable( this.state.endDate ),
+		} );
+
+		// Close the popover
```

Comment:
Yes I also noticed this and was confused. I need to re-follow the code flow to decide what is going on here. When I do I will either

1. Rename the method
2. Create a better method
3. Write a _very_ clear comment.

Reply:
I've resolved this in https://github.com/Automattic/wp-calypso/pull/29938 by created dedicated methods.

Note however, that `close` still reverts the dates because if you close the `Popover` without clicking "Apply" then we need to reset the dates back to the original values. If the user clicks Apply the `close` method will still trigger `revertDates` but by this point the dates have been committed and so it has no effect.